screens/modals/QuantityModal.js

```diff
@@ -0,0 +1,162 @@
+import { FontAwesome5 } from '@expo/vector-icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Modal, TouchableOpacity, View } from 'react-native';
+import Colors from '../../assets/Colors';
+import { Body, ButtonLabel, Title } from '../../components/BaseComponents';
+import LineItemCard from '../../components/LineItemCard';
+import ProductDisplayCard from '../../components/ProductDisplayCard';
+import {
+  ModalCenteredOpacityLayer,
+  ModalContentContainer,
+  ModalCopyContainer,
+  QuantityInput
+} from '../../styled/modal';
+import { ColumnContainer, RoundedButtonContainer } from '../../styled/shared';
+
+export default class QuantityModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      product: null,
+      // Needs to be string type because it's TextInput
+      currentQuantity: '',
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    this.resetCurrentQuantity();
+    this.setState({
+      product,
+      isLoading: false
+    });
+  }
+
+  // Forces a re-render when new props are passed
+  componentWillReceiveProps(nextProps) {
+    const newQuantity = nextProps.product.quantity;
+    if (this.state.currentQuantity !== newQuantity) {
+      this.setState(prevState => ({
```

Comment:
It seems like you’re doing a very similar thing in `componentDidMount` and `componentWillReceiveProps` right now: taking props and setting up state with it. You should keep that as one function and then call that function in both of the lifecycle methods.

Pretty common practice, especially when you need to make API call to update based on new props and all, 1 function for the call, call it in both lifecycle methods